streamz.js

```diff
@@ -152,18 +152,30 @@ Streamz.prototype._fn = function(d) {
   this.push(d);
 };
 
-Streamz.prototype._finalize = noop;
+Streamz.prototype._finalize = function() {
+  // In node 14 we need to look at `_writableState.buffered.length` - in older versions we look at `_writableState.length`
+  if (this.endedCb && !this._concurrent && (this._writableState.buffered && !this._writableState.buffered.length || !this._writableState.length)) {
+    stream.Transform.prototype.end.apply(this, undefined, this.endedCb);
```

Comment:
i don't see a world where this is a problem - but:

```
let endedCb = this.endedCb;
this.endedCB = noop;
stream.Transform.prototype.end.apply(this, undefined, endedCb);
```
would be technically more correct.  It's possible (but unlikely) the `end` method somehow modifies `endedCb`

Reply:
since we are passing the actual `this.endedCb` but not a reference I agree this shouldn't be a problem, but good point though!